### DISCUSSION TITLE
Textures rescaling relative to UV maps

### DISCUSSION THREAD

**Discussion by 01000001khan** [google/model-viewer#3148] [Q&A]
### Description

The problem that I am experiencing is that when applying a new texture to a model, the texture seems to be applied using the same pixel scaling as the original image on the model.  So when applying the same image that is already there, it works perfectly but if you apply a differently sized image, it will appear on only part of the model.  Usually the width and height of UVs are 1 and 1 so that any size or aspect ratio of texture works with UVs and they will always be the same relatively, but it is not working properly now.  I did an experiment in Blender by playing around with the texture and found something which, to me, seems to be proof of exactly what is happening.

This issue appeared after a couple months of not making any changes, so the problem is definitely model-viewer.

Console, along with UV issue.
![image](https://user-images.githubusercontent.com/87445503/150645747-23094787-7812-4fe5-97dd-f4ada708c0e2.png)

Blender experiment
![image](https://user-images.githubusercontent.com/87445503/150645578-6c3d8f8c-67e1-4e3e-a87b-b3d0c0f9c7b5.png)

Expected Result, which was working on model-viewer up until a while ago
![image](https://user-images.githubusercontent.com/87445503/150645848-3b417eca-0951-4e4d-adec-096d21a528c0.png)

#### Live Demo

https://01000001khan.github.io/AR-Speakers/founder.html?

switch the material using the drop down to Non Existent testing material to see the bug.

### Version

- model-viewer: v1.9.2 is the last working version; everything after that will not work

### Browser Affected

- [x] Chrome, version: 97.0.4692.99 (Official Build) (64-bit)
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [x] MacOS
- [x] Windows

**Comment by elalish** [comment]
I'm seeing  lots of big console errors on your site. Try fixing those and see if that helps with the problem you're seeing. I  don't  think the issue is at the renderer level.

**Reply by 01000001khan** [reply]
There were no errors a few months ago, and the main issue started without me having changed anything in that time.  I tried fixing the errors earlier but was not able to do so.  I had quite a hard time getting it to work properly in the first place, so I'm extremely confused now.

Also, on this page (https://01000001khan.github.io/AR-Speakers/index.html?) the animations are broken, but the file hasn't changed.  All the components used to move, but now a few of them are static.  The top model-viewer element is set to autoplay, the other one can be controlled manually via the slider.  Both are broken identically.

All three of these problems, which appeared without any changes to my project, lead me to believe that a change to model-viewer is what caused the issues.

**Reply by elalish** [reply]
I suppose that's possible, and you can easily test  it by selecting a specific version instead of the latest e.g. `src="https://unpkg.com/@google/model-viewer@1.9.1/dist/model-viewer.min.js"`

Please let us know between which versions it appears to have broken.

**Reply by 01000001khan** [reply]
I apologize for the extremely late reply.  The issue appears between version `1.9.2` and version `1.10.0`.  For now, I have pinned my site's code to version `1.9.2` because that's the latest one that's working.

**Reply by elalish** [reply]
You can also try the latest, v1.11. Can you try making a simpler version to see where the problem is? Often the trouble is people writing more script than they actually need; if something simple doesn't work, that'll be a better place for us to look at debug.

**Reply by 01000001khan** [reply]
@elalish Hello again.  I've simplified the code to the minimum amount of stuff necessary to do what I want (with a little bit of formatting).  You can see the code here: https://pastebin.com/d0enJ11Z  I have pinned this code to version 1.9.2, so that you can see the intended behavior, and then if you change it to use the newest version, it will immediately break.  The code uses absolute URLs so you can see the issue by just pasting the code into http://www.csgnetwork.com/htmlcodetest.html

I believe this issue is related to https://github.com/google/model-viewer/issues/1386

**Reply by elalish** [reply]
Ah, thank you; the console has a WebGL error: offset overflows texture dimensions. That seems to be the root of the problem. I notice you're using `setURI`, which I'm considering deprecating in favor of our newer `createTexture` and `setTexture` APIs: https://modelviewer.dev/examples/scenegraph/#createTexturesExample

Could you try those and see if they work better?

**Reply by 01000001khan** [reply]
Thank you very much!  After a little trial and error I got it to work and the scaling issue is no longer a problem.  I'll have to figure out the rest of the stuff again since the whole system is different now, but it is working, which is great :D.  Thank you for your help and patience over the past 6 months.  Here's the working code: https://pastebin.com/QyuB3gi4
